bin/git-obliterate

```diff
@@ -9,15 +9,15 @@ do
     file="$file"' '"$i"
     shift
 done
-test -n "$*" && range="$*"
+test -n "$*" && range=("$@")
 
 test -z "$file" && echo "file required." 1>&2 && exit 1
-if [ -z "$range" ]
+if [ -z "${range[*]}" ]
 then
     git filter-branch -f --index-filter "git rm -r --cached ""$file"" --ignore-unmatch" \
         --prune-empty --tag-name-filter cat -- --all
 else
-    # don't quote $range so that we can forward multiple rev-list arguments
+    # $range is an array so that we can forward multiple rev-list arguments
     git filter-branch -f --index-filter "git rm -r --cached ""$file"" --ignore-unmatch" \
-        --prune-empty --tag-name-filter cat -- $range
+        --prune-empty --tag-name-filter cat -- "${range[@]}"
```

Comment:
Should be "${range[*]}"?

Reply:
This is what [SC2086](https://www.shellcheck.net/wiki/SC2086) doc indicated and looks correct based on the [bash](https://www.gnu.org/software/bash/manual/html_node/Arrays.html) docs. 
> If the subscript is ‘@’ or ‘*’, the word expands to all members of the array name. These subscripts differ only when the word appears within double quotes. If the word is double-quoted, ${name[*]} expands to a single word with the value of each array member separated by the first character of the IFS variable, and ${name[@]} expands each element of name to a separate word.

And indeed that seems to be the case:
```
$ ranges=( "range 1" "range 2" "range 3" ); for range in "${ranges[*]}"; do echo "$range"; done
range 1 range 2 range 3
$ ranges=( "range 1" "range 2" "range 3" ); for range in "${ranges[@]}"; do echo "$range"; done
range 1
range 2
range 3
```

Reply:
Got it